File: HackerRank/Practice/Algorithms/Warmup/MinimaxSum/java/MinimaxSum.java
/*
 * Solution to the "Mini-Max Sum" challenge from HackerRank:
 * https://www.hackerrank.com/challenges/mini-max-sum/problem
 * Created on: 23.03.2020
 * Last modified: 23.03.2020
 * Author: Adam Graliński (adam@gralin.ski)
 * License: MIT
 */

import java.io.*;
import java.util.*;

public class MinimaxSum
{
  public static final int SIZE = 5;
  private static final Scanner scanner = new Scanner(System.in);

  public static void main(String[] args) {
    long input[] = new long[SIZE];
    for (int i = 0; i < SIZE; ++i) {
      input[i] = scanner.nextInt();
    }
    Arrays.sort(input);
    long min = 0, max = 0;
    for (int i = 0; i < SIZE - 1; ++i) {
      min += input[i];
      max += input[i + 1];
    }

    System.out.format("%d %d\n", min, max);
  }
}

File: HackerRank/Practice/Algorithms/Warmup/MinimaxSum/cpp/MinimaxSum.cpp
/**
 * Solution to the "Mini-Max Sum" challenge from HackerRank:
 * https://www.hackerrank.com/challenges/mini-max-sum/problem
 * Created on: 23.03.2020
 * Last modified: 23.03.2020
 * Author: Adam Graliński (adam@gralin.ski)
 * License: MIT
 */

#include <algorithm>
#include <array>
#include <iostream>
#include <numeric>

const std::size_t SIZE = 5;

int main(int, char**)
{
  std::array<long long, SIZE> input;
  for (std::size_t i = 0; i < SIZE; ++i) {
    std::cin >> input[i];
  }
  std::sort(input.begin(), input.end());

  long long min = std::accumulate(input.begin(), input.end() - 1, 0LL);
  long long max = std::accumulate(input.begin() + 1, input.end(), 0LL);
  std::cout << min << " " << max << "\n";
  return 0;
}
